Task #2 | Nijat Amrastanov

namrastanov opened the pull request:
Requirements were to refactor the code and apply best practices to design the application.

- Core, DataAccess, DbMigrations, Tests projects were added
- The Main project was renamed Web
- Introduced MediatR and moved business logic from Controller to the commands and queries
- Introduced Repository pattern to be able to not have the Business layer be dependent on the DataAccess layer
- Introduced FluentValidation
- Introduced Swagger docs
- Introduced AutoMapper
- Added tests for the business operations

Here you can see the solution structure:
![Screenshot_2](https://github.com/ebushuev/VelvetechTestTask/assets/8703157/f1810f9c-9f2c-4b31-a32d-5dff1d926281)

Repository implementations moved to the DataAccess layer, so we do not depend on the DbContext at all in the Core layer:
![Screenshot_3](https://github.com/ebushuev/VelvetechTestTask/assets/8703157/20487dcc-2506-42d6-a9c5-a3922d893ce4)

Test:
![Screenshot_1](https://github.com/ebushuev/VelvetechTestTask/assets/8703157/63d47c4d-6462-4d35-9c64-c09a7fb7cfe9)

Swagger works as expected:
![image](https://github.com/ebushuev/VelvetechTestTask/assets/8703157/fb99121e-2212-4a47-bbcd-b842751543d0)

namrastanov committed:
base structure changes and swagger added

namrastanov committed:
- The Main project was renamed Web
- Introduced MediatR and moved business logic from Controller to the commands and queries
- Introduced Repository pattern to be able to not have the Business layer be dependent on the DataAccess layer
- Introduced FluentValidation
- Introduced Swagger docs
- Introduced AutoMapper

namrastanov committed:
added migrations, added business operation tesst

namrastanov committed:
moved readme file

namrastanov committed:
fixes in Readme file

amrastanov-nitka commented:
Forgot to mention:
- Upgraded .Net version to 6
- Removed redundant property of TodoItem entity.
- Removed the ability to set the IsCompleted for TodoItem on create. There is no sense to set IsCompleted when we create the TodoItem.

namrastanov commented:
Forgot to mention:

- Upgraded .Net version to 6
- Removed redundant property of TodoItem entity.
- Removed the ability to set the IsCompleted for TodoItem on create. There is no sense to set IsCompleted when we create the TodoItem.
